tests/test_process.py

```diff
@@ -90,6 +91,220 @@ def test_upload_dataversion_metadata_without_team_images_id(self, syn: Any):
         )
 
 
+class TestApplyCustomTransformations:
+    """Test the apply_custom_transformations function."""
+
+    @pytest.fixture(scope="function", autouse=True)
+    def standard_transform_function(
+        self,
+    ) -> Callable[[pd.DataFrame, str, Dict[str, pd.DataFrame]], pd.DataFrame]:
+        """mock simple transform function that uses standard parameters such as dataframe, dataset_name, datasets"""
+
+        def _standard_transform_function(
+            df: pd.DataFrame, dataset_name: str, datasets: Dict[str, pd.DataFrame]
+        ) -> pd.DataFrame:
+            """mock simple transform function"""
+            datasets["new_key"] = 2
+            return df.assign(test_col=2, new_key=dataset_name + "_new_key")
+
+        return _standard_transform_function
+
+    @pytest.fixture(scope="function", autouse=True)
+    def special_transform_function(
+        self,
+    ) -> Callable[[pd.DataFrame, int, Dict[str, pd.DataFrame]], pd.DataFrame]:
+        """mock transform function that uses additional parameters"""
+
+        def _mock_transform_with_args(
+            df: pd.DataFrame, test_threshold: int
+        ) -> pd.DataFrame:
+            """mock transform function that takes test_threshold as an argument"""
+            return df.assign(new_key=test_threshold)
+
+        return _mock_transform_with_args
+
+    @pytest.mark.parametrize(
+        "example_config_with_custom_transform,expectation",
+        [
+            (
+                {
```

Comment:
For readability, can you add a comment above each config test case here to explain what thing in the config causes a failure?

Reply:
I added ids to these tests. Does that help?

Reply:
It does help, but not for documentation purposes. The IDs are buried below the configs so you have to look for them and count back to find which one they belong to, and they don't tell me why those configs fail. Comments like `# Fails because mock_transform_function_false doesn't exist` or `# Fails because <x function> doesn't take <y parameters>`  would be really helpful.

Reply:
Also, do the IDs get used anywhere? I'm not sure if pytest magic prints them out on pass/fail or not.